packages/launchpad/src/App.vue

```diff
@@ -1,18 +1,66 @@
 <template>
-  <Layout v-slot="{ item }">
-    <Wizard v-if="item === 'projectSetup'" />
-    <SettingsPage v-if="item === 'settings'" />
-    <RunsPage v-if="item === 'runs'" />
-  </Layout>
+  <div v-if="!backendInitialized || !query.data.value">
+    Loading...
+  </div>
+
+  <div v-else>
+    <Layout v-slot="{ item }">
+      <Wizard 
+        v-if="item === 'projectSetup'" 
+        :query="query.data.value"
+      />
+      <SettingsPage v-if="item === 'settings'" />
+      <RunsPage v-if="item === 'runs'" />
+    </Layout>
+  </div>
 </template>
 
 <script lang="ts" setup>
-import { provideClient } from '@urql/vue'
-import { makeUrqlClient } from './graphql/urqlClient'
+import { computed } from 'vue'
+import { gql, useQuery } from '@urql/vue'
 import Layout from "./layouts/Layout.vue"
 import SettingsPage from './settings/SettingsPage.vue'
 import Wizard from "./setup/Wizard.vue"
+import { AppQueryDocument } from './generated/graphql'
 import RunsPage from "./runs/RunsPage.vue"
 
-provideClient(makeUrqlClient())
+gql`
+query AppQuery {
+  ...Wizard
+  app {
+    activeProject {
+      __typename
+    }
+  }
+}
+`
+
+/**
+ * Sometimes the electron app can start before the GraphQL
+ * server and current project has been initialized.
+ * We poll until those conditions are met, then render the app
+ */
+const query = useQuery({ 
+  query: AppQueryDocument,
+  requestPolicy: 'network-only'
+})
+
+
+let interval: number
+
+const poll = () => {
+  try {
+    if (backendInitialized.value) {
+      window.clearInterval(interval)
+    } else {
+      query.executeQuery()
+    }
+  } catch (e) {
+    // probably graphql server has not initialized yet
+  }
+}
+
+interval = window.setInterval(poll, 200)
```

Comment:
Interesting, I wonder if there's a better way we should be handling things on the server so this race condition isn't a frontend concern.

Reply:
I'd say so. If some activities need to occur before the UI is rendered, I suppose something like

```js
async function start (openElectron) {
  // do stuff
  await something()

  openElectron()
}
```

I'd say once we untangle the mess that is `open_project`, this problem will natural disappear.